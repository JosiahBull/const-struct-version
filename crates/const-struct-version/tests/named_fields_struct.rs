#![allow(dead_code)]

use const_struct_version::StructVersion;

// TODO: Eventually I'd like named fields to be sorted by field name, so they are consistent across
// differnet orderings.

#[derive(StructVersion)]
struct NamedFieldsStruct {
    a: u32,
    b: f32,
}

#[derive(StructVersion)]
struct NamedFieldsStruct2 {
    b: f32,
    a: u32,
}

#[test]
fn test_that_versions_are_different() {
    let version = <NamedFieldsStruct as StructVersion>::version();
    let version2 = <NamedFieldsStruct2 as StructVersion>::version();

    insta::assert_debug_snapshot!(version);
    insta::assert_debug_snapshot!(version2);

    // Ensure all are different.
    assert_ne!(version, version2);
}
