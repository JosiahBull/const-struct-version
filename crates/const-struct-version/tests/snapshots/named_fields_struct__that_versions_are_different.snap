---
source: crates/const-struct-version/tests/named_fields_struct.rs
expression: version
---
"ea02a3d105c8428c7dd5b24e05548548b41aa72e"
